Author: Alastair McKinstry <mckinstry@debian.org>
Description: Fix for bug: incorrect result given for <no> selected by ESC pressed in whiptail
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=#665784
Last-Updated: 2014-06-11
Forwarded: no

Index: newt-0.52.19/dialogboxes.c
===================================================================
--- newt-0.52.19.orig/dialogboxes.c
+++ newt-0.52.19/dialogboxes.c
@@ -572,7 +572,7 @@ int messageBox(const char * text, int he
 
 	answer = newtFormGetCurrent(form);
 
-	if (answer == no)
+	if ((rc == DLG_OKAY) && (answer == no))
 	    rc = DLG_CANCEL;
     }
     else {
